metrics-core/src/test/java/com/yammer/metrics/stats/tests/UniformTimeWindowedSampleTest.java

```diff
@@ -0,0 +1,169 @@
+
+public class UniformTimeWindowedSampleTest {
+  @Test
+  @SuppressWarnings("unchecked")
+  public void aSampleOf100OutOf1000Elements() throws Exception {
+    final UniformTimeWindowedSample sample = new UniformTimeWindowedSample(100, 4);
+    for (int i = 0; i < 1000; i++) {
+      sample.update(i);
+    }
+
+    assertThat("the sample has a size of 25",
+        sample.size(),
+        is(25));
+
+    final Snapshot snapshot = sample.getSnapshot();
+
+    assertThat("the sample has 25 elements",
+        snapshot.size(),
+        is(25));
+
+    for (double i : snapshot.getValues()) {
+      assertThat("the sample only contains elements from the population",
+          i,
+          is(allOf(
+              lessThan(1000.0),
+              greaterThanOrEqualTo(0.0)
+          )));
+    }
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void aSampleOf100OutOf10Elements() throws Exception {
+    final UniformTimeWindowedSample sample = new UniformTimeWindowedSample(100, 3);
+    for (int i = 0; i < 10; i++) {
+      sample.update(i);
+    }
+
+    final Snapshot snapshot = sample.getSnapshot();
+
+    assertThat("the sample has a size of 10",
+        snapshot.size(),
+        is(10));
+
+    assertThat("the sample has 10 elements",
+        snapshot.size(),
+        is(10));
+
+    for (double i : snapshot.getValues()) {
+      assertThat("the sample only contains elements from the population",
+          i,
+          is(allOf(
+              lessThan(10.0),
+              greaterThanOrEqualTo(0.0)
+          )));
+    }
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void aHeavilyBiasedSampleOf100OutOf1000Elements() throws Exception {
+    final UniformTimeWindowedSample sample = new UniformTimeWindowedSample(1000, 3);
+    for (int i = 0; i < 100; i++) {
+      sample.update(i);
+    }
+
+
+    assertThat("the sample has a size of 100",
+        sample.size(),
+        is(100));
+
+    final Snapshot snapshot = sample.getSnapshot();
+
+    assertThat("the sample has 100 elements",
+        snapshot.size(),
+        is(100));
+
+    for (double i : snapshot.getValues()) {
+      assertThat("the sample only contains elements from the population",
+          i,
+          is(allOf(
+              lessThan(100.0),
+              greaterThanOrEqualTo(0.0)
+          )));
+    }
+  }
+
+  @Test
+  public void longPeriodsOfInactivityShouldNotCorruptSamplingState() {
+    final ManualClock clock = new ManualClock();
+    final UniformTimeWindowedSample sample = new UniformTimeWindowedSample(100, 4, clock);
+
+    // add 1000 values at a rate of 10 values/second
+    for (int i = 0; i < 1000; i++) {
+      sample.update(1000 + i);
+      clock.addMillis(100);
+    }
+    assertThat("the sample has 50 elements", sample.getSnapshot().size(), is(50));
+    assertAllValuesBetween(sample, 1000, 2000);
+
+    // wait for 15 hours and add another value.
+    // this should clear all previous values.
+    clock.addHours(15);
```

Comment:
I would add an intermediate check (or perhaps a different test) that checks that a snapshot created well after everything should have expired returns nothing.